libs/deal-ticket/src/components/deal-ticket/deal-ticket-stop-order.spec.tsx

```diff
@@ -176,143 +226,164 @@ describe('StopOrder', () => {
+      if (ocoValue) {
+        await userEvent.click(screen.getByTestId(oco));
+      }
+      await userEvent.click(screen.getByTestId(submitButton));
+      const getByTestId = (id: string) =>
+        screen.getByTestId(ocoPostfix(id, ocoValue));
+      const queryByTestId = (id: string) =>
+        screen.queryByTestId(ocoPostfix(id, ocoValue));
+      expect(getByTestId(priceErrorMessage)).toBeInTheDocument();
+      await userEvent.type(getByTestId(priceInput), '0.001');
+      expect(getByTestId(priceErrorMessage)).toBeInTheDocument();
+
+      // switch to market order type error should disappear
+      await userEvent.click(getByTestId(orderTypeTrigger));
+      await userEvent.click(getByTestId(orderTypeMarket));
+      await userEvent.click(getByTestId(submitButton));
+      expect(queryByTestId(priceErrorMessage)).toBeNull();
+
+      // switch back to limit type
+      await userEvent.click(getByTestId(orderTypeTrigger));
+      await userEvent.click(getByTestId(orderTypeLimit));
+      await userEvent.click(getByTestId(submitButton));
+      expect(getByTestId(priceErrorMessage)).toBeInTheDocument();
+
+      // to small value should be invalid
+      await userEvent.type(getByTestId(priceInput), '0.001');
+      expect(getByTestId(priceErrorMessage)).toBeInTheDocument();
+
+      // clear and fill using valid value
+      await userEvent.clear(getByTestId(priceInput));
+      await userEvent.type(getByTestId(priceInput), '0.01');
+      expect(queryByTestId(priceErrorMessage)).toBeNull();
+    });
   });
 
   it('validates trigger price field', async () => {
     render(generateJsx());
-
-    await userEvent.click(screen.getByTestId(submitButton));
-    expect(screen.getByTestId(triggerPriceErrorMessage)).toBeInTheDocument();
-
-    // switch to trailing percentage offset trigger type
-    await userEvent.click(screen.getByTestId(triggerTypeTrailingPercentOffset));
-    expect(screen.queryByTestId(triggerPriceErrorMessage)).toBeNull();
-
-    // switch back to price trigger type
-    await userEvent.click(screen.getByTestId(triggerTypePrice));
-    expect(screen.getByTestId(triggerPriceErrorMessage)).toBeInTheDocument();
-
-    // to small value should be invalid
-    await userEvent.type(screen.getByTestId(triggerPriceInput), '0.001');
-    expect(screen.getByTestId(triggerPriceErrorMessage)).toBeInTheDocument();
-
-    // clear and fill using value causing immediate trigger
-    await userEvent.clear(screen.getByTestId(triggerPriceInput));
-    await userEvent.type(screen.getByTestId(triggerPriceInput), '0.01');
-    expect(screen.queryByTestId(triggerPriceErrorMessage)).toBeNull();
-    expect(
-      screen.queryByTestId(triggerPriceWarningMessage)
-    ).toBeInTheDocument();
-
-    // change to correct value
-    await userEvent.type(screen.getByTestId(triggerPriceInput), '2');
-    expect(screen.queryByTestId(triggerPriceWarningMessage)).toBeNull();
+    [false, true].forEach(async (ocoValue) => {
```

Comment:
Perhaps use it.each, you'll be able to construct a better test name and therefore better reporting if it fails (same for the other tests)